File: client/src/api/orders/OrderDto.ts
export interface OrderDto {
  id: string;
  number: number;
  userId: string;
  restaurantId: string;
  subtotal: number;
  deliveryFee: number;
  serviceFee: number;
  status: OrderStatus;
  address: string;
  placedAt: string;
  restaurantName: string;
  restaurantAddress: string;
  restaurantPhoneNumber: string;
  paymentIntentClientSecret: string;
  items: OrderItemDto[];
}

export type OrderStatus = "Placed" | "PaymentConfirmed";

export interface OrderItemDto {
  menuItemId: string;
  menuItemName: string;
  menuItemDescription: string;
  menuItemPrice: number;
  quantity: number;
}

File: server/src/Web/Features/Orders/OrderDto.cs
using System;
using System.Collections.Generic;

namespace Web.Features.Orders.GetActiveOrder
{
    public record OrderDto
    {
        public string Id { get; init; }
        public int Number { get; init; }
        public Guid UserId { get; init; }
        public Guid RestaurantId { get; init; }
        public decimal Subtotal { get; init; }
        public decimal DeliveryFee { get; init; }
        public decimal ServiceFee { get; init; }
        public string Status { get; init; }
        public string Address { get; init; }
        public DateTime PlacedAt { get; init; }
        public DateTime? ConfirmedAt { get; init; }
        public string RestaurantName { get; init; }
        public string RestaurantAddress { get; init; }
        public string RestaurantPhoneNumber { get; init; }
        public string PaymentIntentClientSecret { get; init; }
        public List<OrderItemDto> Items { get; init; } = new();
    }

    public record OrderItemDto
    {
        public Guid MenuItemId { get; init; }
        public string MenuItemName { get; init; }
        public string MenuItemDescription { get; init; }
        public decimal MenuItemPrice { get; init; }
        public int Quantity { get; init; }
    }
}
